motoman_mh_support/urdf/mh80_macro.xacro

```diff
@@ -0,0 +1,182 @@
+    </collision>
+  </link>
+  <link name="${prefix}link_l">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/visual/link_l.stl"/>
+      </geometry>
+      <material name="blue"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_l.stl"/>
+      </geometry>
+      <material name="collision_material"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_u">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/visual/link_u.stl"/>
+      </geometry>
+      <material name="blue"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_u.stl"/>
+      </geometry>
+      <material name="collision_material"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_r">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/visual/link_r.stl"/>
+      </geometry>
+      <material name="blue"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_r.stl"/>
+      </geometry>
+      <material name="collision_material"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_b">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/visual/link_b.stl"/>
+      </geometry>
+      <material name="blue"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_b.stl"/>
+      </geometry>
+      <material name="collision_material"/>
+    </collision>
+  </link>
+  <link name="${prefix}link_t">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/visual/link_t.stl"/>
+      </geometry>
+      <material name="blue"/>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0"/>
+      <geometry>
+        <mesh filename="package://motoman_mh_support/meshes/mh80/collision/link_t.stl"/>
+      </geometry>
+      <material name="collision_material"/>
+    </collision>
+  </link>
+  <link name="${prefix}tool0"/>
+  <!-- end of link list -->
+
+  <!-- joint list -->
+  <joint name="${prefix}joint_s" type="revolute">
+    <origin xyz="0 0 0.540" rpy="0 0 0"/>
+    <parent link="${prefix}base_link"/>
+    <child link="${prefix}link_s"/>
+    <axis xyz="0 0 1"/>
+    <limit lower="-3.14159" upper="3.14159" effort="100" velocity="2.967" />
```

Comment:
In more recent models we usually use the degree value with the ${radians(360)}

Effort for the MH80 are:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/marcier/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/marcier/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:11.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">



Axis | Max   Acc/Dec Torque[Nm]
-- | --
1(S) | 4903.32
2(L) | 5883.99
3(U) | 2986.13
4(R) | 772.27
5(B) | 582.52
6(T) | 250.07



</body>

</html>

Reply:
Thanks for the contribution.   Above comment are not essential.  The model is functional and matches MotoSim model.

Reply:
@EricMarcil I can't seem to figure out why the build failed.